pandas/tests/extension/test_categorical.py

```diff
@@ -282,6 +282,16 @@ def _compare_other(self, s, data, op_name, other):
             with pytest.raises(TypeError, match=msg):
                 op(data, other)
 
+    def test_not_equal_with_na(self):
```

Comment:
do we get this right in cases with e.g. datetime64 or datetime64tz categories?

Reply:
It seems to work, added some parameterization over other category types